src/model/range.js

```diff
@@ -462,13 +462,13 @@ export default class Range {
 		} else {
 			const sourceRange = Range.createFromPositionAndShift( sourcePosition, howMany );
 
-			// Edge case for merge detla.
+			// Edge case for merge delta.
 			if (
 				deltaType == 'merge' &&
 				this.isCollapsed &&
 				( this.start.isEqual( sourceRange.start ) || this.start.isEqual( sourceRange.end ) )
 			) {
-				// Collapsed range is in merged element.
+				// Collapsed range is in merged element, at the beginning or at the end of it.
```

Comment:
> Where's the "at the beginning" part checked in the code?

`this.start.isEqual( sourceRange.start )`

Or do you mean tests?

> Actually, the whole condition got too complex and is hard to read.

I haven't changed it. Honestly there's not much to simplify here, except of the last part. I can make some kind of `isAtTheBeginningOrEndOfRange( this.start, sourceRange )` or `isRangeBoundary( this.start, sourceRange )` but IDK how much it helps.